ui/js/controllers/perf/compare.js

```diff
@@ -3,30 +3,30 @@
 perf.controller('CompareChooserCtrl', [
     '$state', '$stateParams', '$scope', 'ThRepositoryModel', 'ThResultSetModel',
     'phCompareDefaultNewRepo', 'phCompareDefaultOriginalRepo', 'JsonPushes',
-    'thPerformanceBranches',
+    'thPerformanceBranches','localStorageService',
     function CompareChooserCtrl($state, $stateParams, $scope,
                                 ThRepositoryModel, ThResultSetModel,
                                 phCompareDefaultNewRepo,
                                 phCompareDefaultOriginalRepo,
-                                JsonPushes, thPerformanceBranches) {
+                                JsonPushes, thPerformanceBranches,
+                                localStorageService) {
         ThRepositoryModel.get_list().success(function(projects) {
             $scope.projects = projects;
             $scope.originalTipList = [];
             $scope.newTipList = [];
-
             $scope.originalProject = _.findWhere(projects, {
-                name: ($stateParams.originalProject ?
-                       $stateParams.originalProject : phCompareDefaultOriginalRepo)
+                name: (localStorageService.get('originalProject') ?
+                       localStorageService.get('originalProject') : phCompareDefaultOriginalRepo)
```

Comment:
This will break the existing code that lets us prepopulate these fields (from $stateParams). e.g. from the "compare this revision in perfherder" link from the job details panel:

We probably need something like this:

```js
function getParameter(paramName, defaultValue) {
  if ($stateParams[paramName]) {
    return $stateParams[paramName];
  } else if (localStorageService.get(paramName) {
    return localStorageService.get(paramName);
  }

  return default;
}
```

Then you could just call that from each of the places. e.g. here:
```js
$scope.originalProject = _.findWhere(projects, {
  name: getParameter('originalProject', phCompareDefaultOriginalRepo),
  ...
});
```

Make sense?